packages/spanner/src/transaction.js

```diff
@@ -283,11 +314,28 @@ Transaction.prototype.commit = function(callback) {
  * @param {function} callback - The callback function.
  */
 Transaction.prototype.request = function(config, callback) {
+  var self = this;
+
   var reqOpts = extend({
     session: this.session.formattedName_
   }, config.reqOpts);
 
-  config.method(reqOpts, callback);
+  var gaxOptions = reqOpts.gaxOptions;
+  delete reqOpts.gaxOptions;
+
+  config.method(reqOpts, gaxOptions, function(err, resp) {
```

Comment:
We might have to assign some type of reference ID to items we've already pushed to the user so we avoid pushing again. Or if the API has a way of starting from a specific token, so it won't respond with those items in the retries. And if that's not the right way to handle it, I think we are limited in solutions, because the point of the stream is to keep memory low and flush response items to the user as quickly as possible. If the request can be invalidated at an arbitrary time before the response is complete, we might need some advice from @vkedia on how this should be handled.